src/components/HeaderAuthOptions/HeaderSignUp/HeaderSignUp.js

```diff
@@ -15,20 +31,80 @@ class HeaderSignUp extends Component {
 
     this.state = {
       email: '',
+      openEmal: false,
+      openPass: false,
+      openUser: false,
+      emailAvail: false,
+      emailValid: false,
+      emailVal: false,
       username: '',
+      usernameVal: false,
+      usernameLength: false,
+      usernameTaken: false,
       identifier: '',
+      submit: false,
       password: '',
-      passwordConfirmation: ''
+      passwordVal: false,
+      passwordLength: false,
+      passwordCapital: false,
+      passwordLower: false,
+      passwordSpecial: false,
+      passwordNumber: false,
+      passwordConfirmation: '',
+      passwordConfirmationVal: false,
+      property: document.documentElement.style.setProperty('--border-show', 'none')
     }
   }
+  // Set state of all fields using helper functions
```

Comment:
I was looking into a cleaner way to do this. The `checkValid()` function checks the state of all the fields at once. I'm looking into just checking the specific fields that are being altered. Maybe a helper function? Or possibly a naming convention that would allow for easier refactoring?